src/Controller/EntityCreateController.php

```diff
@@ -0,0 +1,155 @@
+<?php
+
+/**
+ * @file
+ * Contains \Drupal\src\Controller\EntityCreateController.
+ */
+
+namespace Drupal\src\Controller;
+
+use Drupal\Core\Controller\ControllerBase;
+use Drupal\Core\Url;
+use Symfony\Component\HttpFoundation\Request;
+
+/**
+ * Generic controller to provide /entity/add and /entity/add/{bundle} routes.
+ */
+class EntityCreateController extends ControllerBase {
+
+  /**
+   * Displays add custom entity links for available types.
+   *
+   * @param string $entity_type_id
+   *   The entity type ID.
+   * @param \Symfony\Component\HttpFoundation\Request $request
+   *   The request.
+   *
+   * @return array
+   *   A render array for a list of the custom entity types that can be added or
+   *   if there is only one custom entity type defined for the site, the function
+   *   returns the custom entity add page for that custom entity type.
+   */
+  public function addPage($entity_type_id, Request $request) {
+    $entity_type = $this->entityTypeManager()->getDefinition($entity_type_id);
+    // Get the storage controller for this entity.
+    $bundle_storage = $entity_type->getBundleEntityType()
+      ? $this->entityTypeManager()->getStorage($entity_type->getBundleEntityType())
+      : NULL;
+    // Load all entity types for this entity definition.
+    $types = $bundle_storage->loadMultiple();
+
+    // Check for existing types.
+    if ($types && count($types) == 1) {
+      $type = reset($types);
+      return $this->addForm($entity_type, $type);
+    }
+    if (count($types) === 0) {
+      return [
+        '#markup' => $this->t('You have not created any @entity_label types yet. Go to the <a href=":url">@entity_label type creation page</a> to add a new @entity_label type.', [
+          '@entity_label' => $entity_type->getLabel(),
+          ':url' => Url::fromRoute('entity.' . $entity_type->getBundleEntityType() . '.add_form')->toString(),
+        ]),
+      ];
+    }
+
+    $build = ['add_links'=>[
+      '#theme' => 'links__help',
```

Comment:
Do we want to use the links template here? Node defines its own that uses dt/dd instead of ul/li, here's the generic Commerce version: https://github.com/commerceguys/commerce/blob/8.x-2.x/templates/commerce-add-list.html.twig